frame/support/src/storage/bounded_vec.rs

```diff
@@ -908,4 +975,31 @@ pub mod test {
 		assert!(b.try_extend(vec![4, 5, 6].into_iter()).is_err());
 		assert_eq!(*b, vec![1, 2, 3]);
 	}
+
+	#[test]
+	fn test_serializer() {
+		let c: BoundedVec<u32, ConstU32<6>> = bounded_vec![0, 1, 2];
+		assert_eq!(serde_json::json!(&c).to_string(), r#"[0,1,2]"#);
+	}
+
+	#[test]
+	fn test_deserializer() {
+		let c: BoundedVec<u32, ConstU32<6>> = serde_json::from_str(r#"[0,1,2]"#).unwrap();
+
+		assert_eq!(c.len(), 3);
+		assert_eq!(c[0], 0);
+		assert_eq!(c[1], 1);
+		assert_eq!(c[2], 2);
+	}
+
+	#[test]
+	fn test_deserializer_failed() {
+		let c: Result<BoundedVec<u32, ConstU32<4>>, serde_json::error::Error> =
+			serde_json::from_str(r#"[0,1,2,3,4,5]"#);
+
+		match c {
+			Err(msg) => assert_eq!(format!("{}", msg), "out of bounds at line 1 column 11"),
+			_ => assert_eq!(c.is_err(), true),
```

Comment:
Rather than asserting that `c.is_err()`, this branch should really be `unreachable!()` altogether, so that if deserializing `"[0,1,2,3,4,5]"` is producing errors that we don't expect, this test would panic.

Reply:
Ok, added explicit error message